todo_app/templates/index.html

```diff
@@ -7,12 +7,66 @@ <h1 class="display-4">To-Do App</h1>
     <p class="lead">Just another to-do app.</p>
   </div>
 
+  {% if request.endpoint == 'index' %}
   <div class="row">
-    <div class="col-md-12">
+    <!-- Display all list items -->
+    <div class="col-md-12 mb-5">
       <h2>Items</h2>
       <ul class="list-group mb-4">
-        <!-- List the todo items here -->
+        {% if all_items is iterable and all_items|length == 0 %}
+        <li class="list-group-item">No to-do items</li>
+        {% endif %}
+        {% for item in all_items %}
+        <li class="list-group-item d-flex align-items-center justify-content-between">
+          <a href="/view-item/{{ item.id }}">
+            {% if item.status == 'Completed' %}
+            <del>{{ item.title }}</del>
+            {% else %}
+            {{ item.title }}
+            {% endif %}
+          </a>
+          <form id="remove-item" method="post" action="/remove-item/{{ item.id }}">
+            <span class="badge {{ 'badge-success' if item.status == 'Completed' else 'badge-secondary' }}">{{ item.status }}</span>
+            <button type="submit" class="btn btn-sm" data-toggle="tooltip" data-placement="right" title="Remove item?">&times;</button>
+          </form>
+        </li>
+        {% endfor %}
       </ul>
     </div>
   </div>
-{% endblock %}
\ No newline at end of file
+  {% endif %}
+  <div class="row fixed-bottom py-2 bg-light border-top">
+    <div class="container">
+      <form id="new-item" method="post" action="/add-item">
+        <div class="input-group">
+          <input type="text" name="title" id="title" class="form-control" placeholder="Item title">
```

Comment:
minor but might be nice to have a `required` attribute here so people don't submit empty tasks

Reply:
Thanks, Josh. I've actioned your suggestions and also added a javascript function to auto focus the input field when the document is ready as I figured this is the primary action for the page, making it a bit more accessible.